var/spack/repos/builtin/packages/py-setuptools/package.py

```diff
@@ -200,27 +110,13 @@ class PySetuptools(Package, PythonExtension):
     depends_on("python@2.7:2.8,3.5:", when="@44", type=("build", "run"))
     depends_on("python@2.7:2.8,3.4:", when="@:43", type=("build", "run"))
 
+    # Uses HTMLParser.unescape
+    conflicts("python@3.9:", when="@:41.0")
+
+    # Uses collections.MutableMapping
+    conflicts("python@3.10:", when="@:40.4.2")
+
     # https://github.com/pypa/setuptools/issues/3661
     conflicts("python@3.12:", when="@:67")
 
     depends_on("py-pip", type="build")
-
-    def url_for_version(self, version):
-        url = "https://files.pythonhosted.org/packages/{0}/s/setuptools/setuptools-{1}-{0}-none-any.whl"
-
-        if version >= Version("45.1.0"):
-            python_tag = "py3"
-        else:
-            python_tag = "py2.py3"
-        return url.format(python_tag, version)
-
-    def install(self, spec, prefix):
```

Comment:
@alalazo converted this from sdist to bdist in #31131, not sure if we should revert that PR unless we know that issue won't come back. Can we instead contribute the patch to setuptools and add the latest release?

Reply:
The patch applies to older version of setuptools